rpcserver.go

```diff
@@ -2974,21 +3046,21 @@ func (r *rpcServer) QueryAssetRoots(ctx context.Context,
 
 	// Attempt to retrieve the transfer universe root.
 	rpcsLog.Debugf("Querying for asset (group) transfer universe root "+
-		"for %v", spew.Sdump(universeID))
+		"for %v", spew.Sdump(uniID))
 
-	universeID.ProofType = universe.ProofTypeTransfer
+	uniID.ProofType = universe.ProofTypeTransfer
 
-	transferRoot, err := r.cfg.UniverseArchive.RootNode(ctx, universeID)
-	if err != nil {
+	transferRoot, transferErr := r.cfg.UniverseArchive.RootNode(ctx, uniID)
+	if transferErr != nil {
 		// Do not return at this point if the error only indicates that
 		// the root wasn't found. We may have found the issuance root
 		// above.
-		if !errors.Is(err, universe.ErrNoUniverseRoot) {
-			return nil, err
+		if !errors.Is(transferErr, universe.ErrNoUniverseRoot) {
+			return nil, transferErr
 		}
 	}
 
-	transferRootRPC, err := marshalUniverseRoot(transferRoot)
+	transferRootRPC, err = marshalUniverseRoot(transferRoot)
```

Comment:
Won't this panic if nothing was found above?

Reply:
Hmm yeah I think the `RootNode` call above would err,  return a real root, or `Root{}`. But `marshalUniverseRoot has an empty nil-ness check that exits early on `Root{}`:

https://github.com/lightninglabs/taproot-assets/blob/e569cf553d324297c303d190678320fcc50b9c10/rpcserver.go#L2751

 And we already hit this as the unknown assets would have no transfer root.